scripts/update_ios_android_dependencies.py

```diff
@@ -0,0 +1,654 @@
+    return repl
+
+  substituted_pairs = []
+  to_update = False
+  for line in lines:
+    substituted_line = re.sub(RE_README_POD_VERSION, replace_pod_line,
+                              line)
+    output_lines.append(substituted_line)
+    if substituted_line != line:
+      substituted_pairs.append((line, substituted_line))
+      to_update = True
+
+  if to_update:
+    print('Updating contents of {0}'.format(readme_filepath))
+    for original, substituted in substituted_pairs:
+      print('(-) ' + original + '(+) ' + substituted)
+
+    if not dryrun:
+      with open(readme_filepath, 'w') as readme_file:
+        readme_file.writelines(output_lines)
+    print()
+
+########## END: iOS pods versions update #####################################
+
+########## Android versions update #############################
+
+# Android gMaven repostiory from where we scan available android packages
+# and their versions
+GMAVEN_MASTER_INDEX = "https://dl.google.com/dl/android/maven2/master-index.xml"
+GMAVEN_GROUP_INDEX = "https://dl.google.com/dl/android/maven2/{0}/group-index.xml"
+
+
+def get_latest_maven_versions():
+  latest_versions = {}
+  response = requests.get(GMAVEN_MASTER_INDEX)
+  index_xml = ElementTree.fromstring(response.content)
+  for index_child in index_xml:
+    group_name = index_child.tag
+    group_path = group_name.replace('.', '/')
+    response = requests.get(GMAVEN_GROUP_INDEX.format(group_path))
+    group_xml = ElementTree.fromstring(response.content)
+    for group_child in group_xml:
+      package_name = group_child.tag.replace('-', '_')
+      package_full_name = group_name + "." + package_name
+      package_version = group_child.attrib['versions'].split(',')[-1]
+      latest_versions[package_full_name] = package_version
+  return latest_versions
+
+
+# Regex to match lines like:
+# 'com.google.firebase:firebase-auth:1.2.3'
+RE_GENERIC_DEPENDENCY_MODULE = re.compile(
+    r"(?P<quote>[\'\"])(?P<pkg>[a-zA-Z0-9._-]+:[a-zA-Z0-9._-]+):([0-9.]+)[\'\"]"
+)
+
+def modify_dependency_file(dependency_filepath, version_map, dryrun=True):
+  """Modify a dependency file to reference the correct module versions.
+
+  Looks for lines like: 'com.google.firebase:firebase-auth:1.2.3'
+  for modules matching the ones in the version map, and modifies them in-place.
+
+  Args:
+    dependency_filename: Relative path to the dependency file to edit.
+    version_map: Dictionary of packages to version numbers, e.g. {
+        'com.google.firebase.firebase_auth': '15.0.0' }
+    dryrun (bool, optional): Just print the substitutions.
+                             Do not write to file. Defaults to True.
+  """
+  logging.debug('Reading dependency file: {0}'.format(dependency_filepath))
+  lines = None
+  with open(dependency_filepath, "r") as dependency_file:
+    lines = dependency_file.readlines()
+  if not lines:
+    logging.fatal('Update failed. ' +
+          'Could not read contents from file {0}.'.format(dependency_filepath))
```

Comment:
I agree that in this case it might make sense but in general the recommended string formatting in Python3 is to use the new style formatting. Using ".format" for consistency.
https://docs.python.org/3/library/stdtypes.html?highlight=sprintf#printf-style-string-formatting